cpm

```diff
@@ -1,216 +1,259 @@
 #!/bin/sh
+# shellcheck shell=sh # Written to be posix compatible
+# shellcheck disable=SC2034 # False Trigger
+
+
+# Created by Will Eccles <will@eccles.dev> in 2020 under the terms of Mozilla Public License Version 2.0
+
+# Maintainer info #
+# UPSTREAM="https://github.com/willeccles/cpm"
+# MAINTAINER_EMAIL="will@eccles.dev"
+# MAINTAINER_NICKNAME="willeccles"
+# MAINTAINER_NAME="Willeccles "
+
+
+###! A Script to detect other package managers and make them easy to use
+###! USAGE :- Defined in README.md
+###! Platform:
+###!  - [*] Linux
+###!    - [*] Archlinux
+###!    - [ ] Arya
+###!    - [*] Debian
+###!    - [ ] Exherbo
+###!    - [*] Fedora
+###!    - [ ] Gentoo
+###!    - [*] Nix
+###!    - [*] Ubuntu
+###!    - [*] Void
+###!  - [*] XNU
+###!    - [*] MacOS
+###! Package Manager:
+###!  - [*] Apt
+###!  - [ ] ALPS
+###!  - [ ] Paludis
+###!  - [*] dnf
+###!  - [*] MacPorts
+###!  - [*] nix
+###!  - [*] Pacman
+###!  - [ ] Portage
+###!  - [*] xbps
+
+
+# FIXME: Add Version
+# UPSTREAM_VERSION=?
+
 
 phi() {
-    >&2 printf "\033[33;1m-> \033[35;1m%s | %s\033[m %s\n" "$1" "$2" "$3"
+	>&2 printf "\033[33;1m-> \033[35;1m%s | %s\033[m %s\n" "$1" "$2" "$3"
 }
 
 usage() {
-    >&2 echo "cpm [i|r|l|C|u|U|s|S|I|c|h] [pkg]..."
-    phi i "install" "install one or more packages"
-    phi r "remove " "remove one or more packages"
-    phi l "list   " "list installed packages"
-    phi C "count  " "count installed packages"
-    phi u "update " "update package lists"
-    phi U "upgrade" "upgrade all packages"
-    phi s "search " "search for a package"
-    phi S "show   " "show information about a package"
-    phi I "info   " "same as show"
-    phi c "clean  " "clean up leftover files/caches/orphans"
-    phi h "help   " "show this message"
+	>&2 echo "cpm [i|r|l|C|u|U|s|S|I|c|h] [pkg]..."
+	phi i "install" "install one or more packages"
+	phi r "remove " "remove one or more packages"
+	phi l "list   " "list installed packages"
+	phi C "count  " "count installed packages"
+	phi u "update " "update package lists"
+	phi U "upgrade" "upgrade all packages"
+	phi s "search " "search for a package"
+	phi S "show   " "show information about a package"
+	phi I "info   " "same as show"
+	phi c "clean  " "clean up leftover files/caches/orphans"
+	phi h "help   " "show this message"
 }
```

Comment:
Other then defining a function to return a docstring using echo is ugly and not portable.. seems as a good idea to replace this with a heredoc instead i.e

```sh
cat <<-EOF 
...
EOF
```